cyvcf2/cyvcf2.pyx

```diff
@@ -1380,7 +1380,7 @@ cdef class Variant(object):
             size *= data.shape[1]
 
         cdef int ret
-        if np.issubdtype(data.dtype, np.int):
+        if np.issubdtype(data.dtype, np.signedinteger) or np.issubdtype(data.dtype, np.unsignedinteger):
             aint = data.astype(np.int32).reshape((size,))
             ret = bcf_update_format_int32(self.vcf.hdr, self.b, to_bytes(name), &aint[0], size)
         elif np.issubdtype(data.dtype, np.float):
```

Comment:
This line `np.issubdtype(data.dtype, np.float)` triggers a numpy 1.14+ deprecation warning (see https://github.com/numpy/numpy/pull/9505)

```py
# python setup.py test
cyvcf2/tests/test_reader.py:586: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  v.set_format("PS", np.array([0.555, 1.111], dtype=np.float32))
```

Changing it to `np.issubdtype(data.dtype, np.floating)` will fix this warning.

Reply:
updated. thanks!